src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -248,22 +259,80 @@ object EdgeProcessor extends StrictLogging {
     result.sequence
   }
 
-  // Temporary for testing join/leave logic.
+  def resolveTransactions(
+    hashes: Seq[String],
+    priorityPeer: Id
+  )(implicit dao: DAO): Future[Seq[TransactionCacheData]] = {
 
-  def handleSignatureRequest(sr: SignatureRequest)(implicit dao: DAO): SignatureResponse = {
-    //if (sr.facilitators.contains(dao.id)) {
-    // val replyTo = sr.checkpointBlock.witnessIds.head
-    val updated = if (sr.checkpointBlock.simpleValidation()) {
-      Some(hashSign(sr.checkpointBlock.baseHash, dao.keyPair))
-    } else {
-      None
+    implicit val exec: ExecutionContextExecutor = dao.signatureExecutionContext
+
+    def lookupTransaction(hash: String, client: APIClient): Future[Option[TransactionCacheData]] = {
+      async {
+        await(
+          client.getNonBlocking[Option[TransactionCacheData]](
+            s"transaction/$hash",
+            timeout = 4.seconds
+          )
+        )
+      }
+    }
+
+    def resolveTransaction(hash: String,
+                           peers: Seq[APIClient]): Future[Option[TransactionCacheData]] = {
+
+      var remainingPeers = peers.tail
+
+      lookupTransaction(hash, peers.head).transformWith {
+        case Success(Some(transactionCacheData)) => Future.successful(Some(transactionCacheData))
+        case _ =>
+          remainingPeers.headOption
+            .map { peer =>
+              remainingPeers = remainingPeers.filterNot(_ == peer)
+              lookupTransaction(hash, peer)
+            }
+            .getOrElse(Future.failed(new Exception("Ran out of peers to query for transaction")))
+      }
+
+    }
+
+    // Change to facilitator ordering
+    val peers = dao.readyPeers.toSeq.sortBy { _._1 != priorityPeer }.map { _._2.client }
```

Comment:
btw -- false sorts lower than true, so I think your non priorityPeer's are going to be in the front here. is that what you want?

Reply:
priorityPeer should be in the front